src/Audio/CueData.cs

```diff
@@ -789,12 +878,23 @@ public uint Timestamp
 			private set;
 		}
 
-		protected static Random random = new Random();
+		public int Count { get; private set; }
```

Comment:
Now with count and frequency available, we could update the timestamp after an event fires to schedule it again.